lib/utils/checkAgainstRule.js

```diff
@@ -3,39 +3,45 @@
 const normalizeRuleSettings = require('../normalizeRuleSettings');
 const Result = require('postcss/lib/result');
 const rules = require('../rules');
+const { isPlainObject } = require('./validateTypes');
 
 /**
  * Useful for third-party code (e.g. plugins) to run a PostCSS Root
  * against a specific rule and do something with the warnings
  * @template T
  * @template {Object} O
  * @param {{
-		ruleName: string,
-		ruleSettings: import('stylelint').ConfigRuleSettings<T, O>,
-		root: import('postcss').Root,
-	}} options
+ *   ruleName: string,
+ *   ruleSettings: import('stylelint').ConfigRuleSettings<T, O>,
+ *   root: import('postcss').Root,
+ *   result?: import('stylelint').PostcssResult,
+ * }} options
  * @param {(warning: import('postcss').Warning) => void} callback
  * @returns {void}
```

Comment:
[suggest] I noticed it's possible to reduce type annotations like this. How about the refactoring?

```js
/**
 * Useful for third-party code (e.g. plugins) to run a PostCSS Root
 * against a specific rule and do something with the warnings.
 *
 * @type {typeof import('stylelint').utils.checkAgainstRule}
 */
function checkAgainstRule(options, callback) {
```

The following line also can be simplified:

https://github.com/stylelint/stylelint/blob/35c0cfa244f725a3548e8ba884edf70c2bc67732/lib/utils/checkAgainstRule.js#L52

Reply:
👀 Good eye! This is much cleaner: https://github.com/stylelint/stylelint/pull/6460/commits/9f600025bc4282df765654c0f22908284bc4210d